ffi/Gosu.cpp

```diff
@@ -1,251 +1,200 @@
-        gosu_window = reinterpret_cast<Gosu::Window *>(window);
-    }
-    return Gosu::available_width(gosu_window);
+    return Gosu::available_width(window);
 }
 
-unsigned Gosu_available_height(Gosu_Window *window)
+GOSU_FFI_API unsigned Gosu_available_height(Gosu_Window* window)
 {
-    Gosu::Window *gosu_window = nullptr;
-    if (window != nullptr) {
-        gosu_window = reinterpret_cast<Gosu::Window *>(window);
-    }
-    return Gosu::available_height(gosu_window);
+    return Gosu::available_height(window);
 }
 
-unsigned Gosu_screen_width(Gosu_Window *window)
+GOSU_FFI_API unsigned Gosu_screen_width(Gosu_Window* window)
 {
-    Gosu::Window *gosu_window = nullptr;
-    if (window != nullptr) {
-        gosu_window = reinterpret_cast<Gosu::Window *>(window);
-    }
-    return Gosu::screen_width(gosu_window);
+    return Gosu::screen_width(window);
 }
 
-unsigned Gosu_screen_height(Gosu_Window *window)
+GOSU_FFI_API unsigned Gosu_screen_height(Gosu_Window* window)
 {
-    Gosu::Window *gosu_window = nullptr;
-    if (window != nullptr) {
-        gosu_window = reinterpret_cast<Gosu::Window *>(window);
-    }
-    return Gosu::screen_height(gosu_window);
+    return Gosu::screen_height(window);
 }
 
-int Gosu_button_down(int btn)
+GOSU_FFI_API int Gosu_button_down(int id)
 {
-    return Gosu::Input::down((Gosu::Button)btn);
+    return Gosu::Input::down(static_cast<Gosu::Button>(id));
 }
 
-const char *Gosu_button_id_to_char(int id)
+GOSU_FFI_API const char* Gosu_button_id_to_char(int id)
 {
     static thread_local std::string button;
-    button = Gosu::Input::id_to_char((Gosu::Button)id);
-
+    button = Gosu::Input::id_to_char(static_cast<Gosu::Button>(id));
     return button.c_str();
 }
 
-unsigned Gosu_button_char_to_id(const char *btn)
+GOSU_FFI_API unsigned Gosu_button_char_to_id(const char* btn)
 {
     return Gosu::Input::char_to_id(btn);
 }
 
-const char *Gosu_button_name(int btn)
+GOSU_FFI_API const char* Gosu_button_name(int id)
 {
     static thread_local std::string name;
-    name = Gosu::Input::button_name((Gosu::Button)btn);
+    name = Gosu::Input::button_name(static_cast<Gosu::Button>(id));
     return name.empty() ? nullptr : name.c_str();
 }
 
-const char *Gosu_gamepad_name(int id)
+GOSU_FFI_API const char* Gosu_gamepad_name(int id)
 {
     static thread_local std::string name;
     name = Gosu::Input::gamepad_name(id);
     return name.empty() ? nullptr : name.c_str();
 }
 
-double Gosu_axis(int btn)
+GOSU_FFI_API double Gosu_axis(int id)
 {
-    return Gosu::Input::axis((Gosu::Button)btn);
+    return Gosu::Input::axis(static_cast<Gosu::Button>(id));
 }
 
-int Gosu_fps()
+GOSU_FFI_API int Gosu_fps()
 {
     return Gosu::fps();
 }
 
-const char *Gosu_language()
+GOSU_FFI_API const char* Gosu_language()
 {
     static thread_local std::string language;
     language = Gosu::language();
-
     return language.c_str();
 }
 
-long Gosu_milliseconds()
+GOSU_FFI_API uint64_t Gosu_milliseconds()
```

Comment:
I've used types with a specific size where it matters (here and in Gosu_Color.hxx); would it make sense to make the size explicit _everywhere_? 🤔